app/controllers/queue_items_controller.rb

```diff
@@ -11,12 +11,43 @@ def create
   end
 
   def destroy
+    queue_item = QueueItem.find_by(id: params[:id], user: current_user)
```

Comment:
The way you suggested is throwing a RecordNotFound exception if the record can't be found. The one i have used returns nil if a record is not found. I just didn't want to rescue an exception when there was an alternative.